allennlp/models/ensemble/bidaf_ensemble.py

```diff
@@ -0,0 +1,163 @@
+        by the average confidence of the span_start and span_end.
+
+        Parameters
+        ----------
+        question : Dict[str, torch.LongTensor]
+            From a ``TextField``.
+        passage : Dict[str, torch.LongTensor]
+            From a ``TextField``.  The model assumes that this passage contains the answer to the
+            question, and predicts the beginning and ending positions of the answer within the
+            passage.
+        span_start : ``torch.IntTensor``, optional
+            From an ``IndexField``.  This is one of the things we are trying to predict - the
+            beginning position of the answer with the passage.  This is an `inclusive` token index.
+            If this is given, we will compute a loss that gets included in the output dictionary.
+        span_end : ``torch.IntTensor``, optional
+            From an ``IndexField``.  This is one of the things we are trying to predict - the
+            ending position of the answer with the passage.  This is an `inclusive` token index.
+            If this is given, we will compute a loss that gets included in the output dictionary.
+        metadata : ``List[Dict[str, Any]]``, optional
+            If present, this should contain the question ID, original passage text, and token
+            offsets into the passage for each instance in the batch.  We use this for computing
+            official metrics using the official SQuAD evaluation script.  The length of this list
+            should be the batch size, and each dictionary should have the keys ``id``,
+            ``original_passage``, and ``token_offsets``.  If you only want the best span string and
+            don't care about official metrics, you can omit the ``id`` key.
+
+        Returns
+        -------
+        An output dictionary consisting of:
+        best_span : torch.IntTensor
+            The result of a constrained inference over ``span_start_logits`` and
+            ``span_end_logits`` to find the most probable span.  Shape is ``(batch_size, 2)``
+            and each offset is a token index.
+        best_span_str : List[str]
+            If sufficient metadata was provided for the instances in the batch, we also return the
+            string from the original passage that the model thinks is the best answer to the
+            question.
+        """
+
+        subresults = []
+        for submodel in self.submodels:
+            # TODO(michaels): should I not call forward directly here and instead call submodel(**)?
+            subresults.append(submodel.forward(question, passage, span_start, span_end, metadata))
+
+        batch_size = len(subresults[0]["best_span"])
+
+        # TODO(michaels): The response here is a tensor rather than a variable (as in BiDAF).  I think that is
+        # ok because this isn't used for training, but I wanted to ask.
+        output = {
```

Comment:
The response here is a tensor rather than a variable (as in BiDAF).  I think that is ok because this isn't used for training, but I wanted to ask.

Reply:
Yeah, probably fine, and easy to fix if we decide we need to fix it later.  Pytorch is changing how tensors / variables work soon, anyway.